packages/apps/docs/src/scripts/importChangelogs/index.ts

```diff
@@ -0,0 +1,220 @@
+import type { IScriptResult } from '@kadena/docs-tools';
+import fs from 'fs';
+import type { Node, Text } from 'mdast';
+import { remark } from 'remark';
+import type { Root } from 'remark-gfm';
+import { clone } from '../importReadme';
+import { deleteTempDir } from '../importReadme/importRepo';
+import { runPrettier } from '../runPrettier';
+import { isParent } from '../utils';
+import {
+  CHANGELOGFILENAME,
+  REPOS,
+  VersionPosition,
+  errors,
+  success,
+} from './constants';
+import { getCommitId } from './utils/commits';
+import { enrichContent } from './utils/enrichContent';
+import { getGitHubData } from './utils/github';
+import { getChangelog, writeContent } from './utils/misc';
+import { getPrId } from './utils/prs';
+
+const getCurrentContentCreator = () => {
+  let content: IChangelogComplete;
+
+  return (): IChangelogComplete => {
+    if (content) return content;
+    const result = fs.readFileSync(CHANGELOGFILENAME, 'utf-8');
+    content = JSON.parse(result.trim());
+    return content ?? {};
+  };
+};
+
+const getCurrentContent = getCurrentContentCreator();
+
+//TESTABLE
+const createVersion = (branch: Node): IChangelogPackageVersion => {
+  return {
+    label: (branch as any).children[0].value ?? '',
+    isLocked: false,
+    authors: [],
+    patches: [],
+    minors: [],
+    miscs: [],
+  };
+};
+
+const crawlContent = (tree: Node): string => {
+  let content = '';
+
+  const innerCrawl = (tree: Node): string => {
+    if (isParent(tree)) {
+      tree.children?.forEach((branch) => {
+        const text = (branch as Text).value;
+        if (text) {
+          content = `${content}${text}`;
+          return content;
+        }
+        return innerCrawl(branch);
+      });
+    }
+    return content;
+  };
+
+  innerCrawl(tree);
+
+  return content;
+};
+
+const createRecord = (content: Node): IChangelogVersionRecord => {
```

Comment:
maybe adding some more context on the method name, it would be better to understand what it does